app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
+    for (var i = 0; i < $scope.hits.length - 1; i++) {
+      $scope.hits[i] = calcDamage($scope.hits[i].hit);
+    }
+    if (!skipSerialization) {
+      serializeData();
+    }
+  };
+
+  $scope.updateHitTable = function (index) {
+    if ($scope.hits[index].hit === null) {
+      $scope.hits.splice(index, 1);
+      return;
+    }
+    if ($scope.hits[$scope.hits.length - 1].hit !== null) {
+      $scope.hits.push({hit: null});
+    }
+    $scope.hits[index] = calcDamage($scope.hits[index].hit);
+    serializeData();
+  };
+
+  $scope.getHitWidth = function (damage) {
```

Comment:
The width we are looking for is the health bar section, which may not be the 'hit' part of the bar, it may be just the remaining life. Maybe getHealthBarSectionWidth()? That's kind of long. I use just 'bar' in places to refer to the subset so maybe getBarWidth()